pkg/operator/rules.go

```diff
@@ -84,51 +108,55 @@ func (prs *PrometheusRuleSelector) generateRulesConfiguration(promRule *monitori
 	for i := range promRuleSpec.Groups {
 		if promRuleSpec.Groups[i].Limit != nil && prs.version.LT(minVersionLimits) {
 			promRuleSpec.Groups[i].Limit = nil
-			level.Warn(logger).Log("msg", fmt.Sprintf("`limit` is supported only from %s version %q", component, minVersionLimits))
+			level.Warn(prs.logger).Log("msg", fmt.Sprintf("`limit` is supported only from %s version %q", component, minVersionLimits))
 		}
+
 		if prs.ruleFormat == PrometheusFormat {
 			// Unset partialResponseStrategy field.
 			promRuleSpec.Groups[i].PartialResponseStrategy = ""
 		}
-	}
 
-	content, err := yaml.Marshal(promRuleSpec)
-	if err != nil {
-		return "", errors.Wrap(err, "failed to marshal content")
-	}
-	errs := ValidateRule(promRuleSpec)
-	if len(errs) != 0 {
-		const m = "Invalid rule"
-		level.Debug(logger).Log("msg", m, "content", content)
-		for _, err := range errs {
-			level.Info(logger).Log("msg", m, "err", err)
+		for j := range promRuleSpec.Groups[i].Rules {
+			switch prs.ruleFormat {
+			case PrometheusFormat:
+				if promRuleSpec.Groups[i].Rules[j].KeepFiringFor != nil && prs.version.LT(minVersionKeepFiringFor) {
+					promRuleSpec.Groups[i].Rules[j].KeepFiringFor = nil
+					level.Warn(prs.logger).Log("msg", fmt.Sprintf("`keep_firing_for` is supported only from %s version %q", component, minVersionKeepFiringFor))
```

Comment:
(suggestion) we can make the log message consistent with the ones from ConfigGenerator.

```suggestion
					level.Warn(prs.logger).Log("msg", fmt.Sprintf("ignoring 'keep_firing_for' not supported by %s", component), "minimum_version", minVersionKeepFiringFor)
```